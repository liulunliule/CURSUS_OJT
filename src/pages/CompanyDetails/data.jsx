import { useState, useEffect } from "react";
import axios from "axios";

export const useFetchOffices = () => {
  const [offices, setOffices] = useState([]);

  useEffect(() => {
    axios.get("https://668f4a0880b313ba09178dee.mockapi.io/api/offices")
      .then(response => {
        setOffices(response.data);
      })
      .catch(error => {
        console.error("Error fetching office data:", error);
      });
  }, []);

  return offices;
};