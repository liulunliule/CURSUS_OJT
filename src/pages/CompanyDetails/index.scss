.company-page {
  font-family: Arial, sans-serif;
  margin-top: 56px;
  color: #333;

  p {
    color: gray;
    margin-bottom: 30px;
  }
  
  h1, h2 {
    color: #333;
    font-weight: bold;
    margin-bottom: 20px;
  }
}

.custom-nav {
  .custom-ul {
    list-style: none;
    display: flex;
    justify-content: center;
    gap: 20px;
    padding: 0;
    margin: 0;

    .custom-li {
      cursor: pointer;
      padding: 10px;
      position: relative;

      &.selected::after {
        content: '';
        position: absolute;
        bottom: -3px;
        left: 0;
        right: 0;
        height: 3px;
        background: red;
      }

      &:hover {
        font-weight: bold;
      }
    }
  }
}

.origins {
  padding: 50px 0;
  background: #f0f0f0;

  .origins-content {
    display: flex;
    justify-content: space-between;
    align-items: center;
    max-width: 1200px;
    margin: 0 auto;
    padding: 0 20px;

    .origins-text {
      flex: 1;

      h2 {
        font-size: 1.5em;
        font-weight: bold;
        text-align: left;
      }

      p {
        max-width: 1000px;
        text-align: left;
        margin-bottom: 30px;
        line-height: 1.6;
      }
    }

    .origins-image {
      flex: 1;
      text-align: center;

      img {
        max-width: 100%;
        height: auto;
      }
    }
  }
}

.offices {
  padding: 50px 0;
  text-align: center;

  p {
    text-align: center;
    margin-bottom: 30px;
  }
}

.offices h2 {
  font-size: 1.5em;
  font-weight: bold;
}

.office-list {
  display: flex;
  justify-content: center;
  gap: 20px;
  flex-wrap: wrap;
}

.office-item {
  max-width: 500px;
  text-align: center;
}

.office-item img {
  width: 394px;
  height: 222px;
  margin-bottom: 10px;
}

h3 {
  color: #333;
  font-weight: bold;
  margin-bottom: 20px;
  text-align: left;
}

h5 {
  color: gray;
  font-size: 15px;
  text-align: left;
}

h6 {
  margin-top: 20px;
  font-size: 15px;
  color: gray;
  text-align: left;
}