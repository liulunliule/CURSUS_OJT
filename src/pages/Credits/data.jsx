import { useState, useEffect } from "react";
import axios from "axios";
import {
    fetchCreditData,
    fetchTransactionsData,
} from "../../services/apiService";

export const useFetchCreditsData = () => {
    const [addedCredits, setAddedCredits] = useState({});
    const [transactions, setTransactions] = useState([]);

    useEffect(() => {
        const fetchAddedCredits = async () => {
            try {
                const response = await fetchCreditData();
                setAddedCredits(response.data[0]);
            } catch (error) {
                console.error("Error fetching added credits data:", error);
            }
        };

        const fetchTransactions = async () => {
            try {
                const response = await fetchTransactionsData();
                setTransactions(response.data);
            } catch (error) {
                console.error("Error fetching transactions data:", error);
            }
        };

        fetchAddedCredits();
        fetchTransactions();
    }, []);

    return { addedCredits, transactions };
};
