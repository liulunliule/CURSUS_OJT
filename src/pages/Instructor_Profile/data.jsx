import axios from "axios";

export const fetchUserInfo = async (targetId) => {
  try {
    const response = await axios.get("https://66751909a8d2b4d072eeb172.mockapi.io/InstructorProfile");
    return response.data.find((item) => item.id === targetId);
  } catch (error) {
    console.error("Error fetching user info:", error);
    throw error;
  }
};

export const fetchUserPosts = async () => {
  try {
    const response = await axios.get("https://6677cf6e0bd45250561c9488.mockapi.io/course");
    return response.data;
  } catch (error) {
    console.error("Error fetching user posts:", error);
    throw error;
  }
};

export const fetchUserReviews = async () => {
  try {
    const response = await axios.get("https://6677cf6e0bd45250561c9488.mockapi.io/reviews");
    return response.data;
  } catch (error) {
    console.error("Error fetching user reviews:", error);
    throw error;
  }
};

export const fetchSubscriptions = async () => {
  try {
    const response = await axios.get("https://6678f8ae0bd4525056207d59.mockapi.io/subscriptions");
    return response.data;
  } catch (error) {
    console.error("Error fetching subscriptions:", error);
    throw error;
  }
};
