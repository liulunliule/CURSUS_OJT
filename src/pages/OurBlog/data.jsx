import { useState, useEffect } from "react";
import axios from "axios";

export const useFetchBlogs = () => {
  const [blogs, setBlogs] = useState([]);

  useEffect(() => {
    axios.get("https://668f4a0880b313ba09178dee.mockapi.io/api/blogs")
      .then((response) => setBlogs(response.data))
      .catch((error) => console.error("Error fetching blog data:", error));
  }, []);

  return blogs;
};