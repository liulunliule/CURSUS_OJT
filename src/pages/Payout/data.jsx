import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchNextPayout, fetchTransactions } from '../../redux/features/payoutSlice';

export const useFetchPayoutData = () => {
  const dispatch = useDispatch();
  const nextPayout = useSelector((state) => state.payout.nextPayout);
  const transactions = useSelector((state) => state.payout.transactions);
  const status = useSelector((state) => state.payout.status);
  const error = useSelector((state) => state.payout.error);

  useEffect(() => {
    if (status === 'idle') {
      dispatch(fetchNextPayout());
      dispatch(fetchTransactions());
    }
  }, [status, dispatch]);

  return { nextPayout, transactions, status, error };
};