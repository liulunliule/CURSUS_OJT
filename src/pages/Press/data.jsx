import { useState, useEffect } from "react";
import axios from "axios";

export const useFetchPressData = () => {
  const [news, setNews] = useState([]);
  const [releases, setReleases] = useState([]);

  useEffect(() => {
    const fetchNews = async () => {
      try {
        const response = await axios.get('https://6690833cc0a7969efd9c4da4.mockapi.io/api/press_news');
        setNews(response.data);
      } catch (error) {
        console.error("Error fetching news data:", error);
      }
    };

    const fetchReleases = async () => {
      try {
        const response = await axios.get('https://6690833cc0a7969efd9c4da4.mockapi.io/api/press_releases');
        setReleases(response.data);
      } catch (error) {
        console.error("Error fetching releases data:", error);
      }
    };

    fetchNews();
    fetchReleases();
  }, []);

  return { news, releases };
};