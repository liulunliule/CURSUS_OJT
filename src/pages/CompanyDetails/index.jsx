import React, { useState } from "react";
import { useFetchOffices } from "./data";
import "./index.scss";

const CompanyDetails = () => {
  const [selected, setSelected] = useState("Company");
  const { offices, status, error } = useFetchOffices();

  const handleSelect = (item) => {
    window.location.href = `/secondLayout/${item.toLowerCase()}`;
    setSelected(item);
  };

  if (status === 'loading') {
    return <div>Loading...</div>;
  }

  if (status === 'failed') {
    return <div>Error: {error}</div>;
  }

  return (
    <div className="company-page">
      <header className="custom-header" style={{ backgroundColor: "#f8f8f8" }}>
        <nav className="custom-nav">
          <ul className="custom-ul">
            {["About", "Blog", "Company", "Careers", "Press"].map((item) => (
              <li
                key={item}
                className={`custom-li ${selected === item ? "selected" : ""}`}
                onClick={() => handleSelect(item)}
              >
                {item}
              </li>
            ))}
          </ul>
        </nav>
      </header>

      <section className="hero" style={{ backgroundColor: "#f8f8f8", padding: "40px" }}>
        <h1 style={{ marginTop: "6px" }}>Expanding learning opportunities</h1>
      </section>

      <section className="origins">
        <div className="origins-content">
          <div className="origins-text">
            <h2>Our Origins</h2>
            <p>
              Cursus was founded in 2020 by computer science instructor with a
              vision to provide anyone, anywhere with access to the worldâ€™s best
              education. Now many instructors put their courses online for
              anyone to take and taught more learners in a few months than they
              could over an entire lifetime in the classroom. Today, Cursus has
              expanded to reach more than 40 million people and 2,300 businesses
              around the world. On Cursus you can find online courses,
              instructors, and certificates from Cursus.
            </p>
          </div>
          <div className="origins-image">
            <img src="https://gambolthemes.net/html-items/cursus-new-demo/images/about/company.jpg" alt="Company Origins" />
          </div>
        </div>
      </section>

      <section className="offices">
        <h2>Our Offices</h2>
        <p>Cursus branches around the world</p>
        <div className="office-list">
          {Array.isArray(offices) && offices.map((office) => (
            <div key={office.Offices_ID} className="office-item">
              <img src={office.Offices_Image} alt={`Office in ${office.Nation}`} />
              <h3>{office.Nation}</h3>
              <h5>{office.Address.split(",")[0]}</h5>
              <h5>{office.Address.split(",").slice(1).join(",")}</h5>
              <h6>{office.Hotline}</h6>
            </div>
          ))}
        </div>
      </section>
    </div>
  );
};

export default CompanyDetails;