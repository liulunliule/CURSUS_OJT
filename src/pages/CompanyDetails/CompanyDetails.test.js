import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import CompanyDetails from './index';
import { useFetchOffices } from './data';

jest.mock('./data', () => ({
  useFetchOffices: jest.fn()
}));

describe('CompanyDetails', () => {
  const officesMock = [
    {
      Offices_ID: 1,
      Offices_Image: 'https://example.com/image1.jpg',
      Nation: 'Country A',
      Address: 'Address A, City A',
      Hotline: '123-456-789'
    },
    {
      Offices_ID: 2,
      Offices_Image: 'https://example.com/image2.jpg',
      Nation: 'Country B',
      Address: 'Address B, City B',
      Hotline: '987-654-321'
    }
  ];

  beforeEach(() => {
    useFetchOffices.mockReturnValue({
      offices: officesMock,
      status: 'succeeded',
      error: null
    });
  });

  test('renders CompanyDetails correctly', () => {
    render(<CompanyDetails />);

    expect(screen.getByRole('heading', { name: /Expanding learning opportunities/i })).toBeInTheDocument();

    ['About', 'Blog', 'Company', 'Careers', 'Press'].forEach(item => {
      expect(screen.getByText(item)).toBeInTheDocument();
    });

    expect(screen.getByText(/Our Origins/i)).toBeInTheDocument();
    expect(screen.getByText(/Cursus was founded in 2020/i)).toBeInTheDocument();

    expect(screen.getByText(/Our Offices/i)).toBeInTheDocument();
    officesMock.forEach(office => {
      expect(screen.getByText(office.Nation)).toBeInTheDocument();
      expect(screen.getByText(office.Address.split(',')[0])).toBeInTheDocument();
      expect(screen.getByText(office.Hotline)).toBeInTheDocument();
    });
  });

  test('navigates to correct page when menu item is clicked', () => {
    render(<CompanyDetails />);

    delete window.location;
    window.location = { href: '' };

    const menuItems = ['About', 'Company', 'Blog', 'Careers', 'Press'];
    menuItems.forEach(item => {
      fireEvent.click(screen.getByText(item));
      expect(window.location.href).toContain(`/secondLayout/${item.toLowerCase()}`);
    });
  });

  test('displays loading state', () => {
    useFetchOffices.mockReturnValue({
      offices: [],
      status: 'loading',
      error: null
    });

    render(<CompanyDetails />);
    expect(screen.getByText('Loading...')).toBeInTheDocument();
  });

  test('displays error state', () => {
    useFetchOffices.mockReturnValue({
      offices: [],
      status: 'failed',
      error: 'Failed to fetch offices'
    });

    render(<CompanyDetails />);
    expect(screen.getByText('Error: Failed to fetch offices')).toBeInTheDocument();
  });
});