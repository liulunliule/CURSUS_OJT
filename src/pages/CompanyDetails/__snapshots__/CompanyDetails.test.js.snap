// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Company Details 1`] = `
<div
  className="company-page"
>
  <header
    className="custom-header"
    style={
      Object {
        "backgroundColor": "#f8f8f8",
      }
    }
  >
    <nav
      className="custom-nav"
    >
      <ul
        className="custom-ul"
      >
        <li
          className="custom-li "
          onClick={[Function]}
        >
          About
        </li>
        <li
          className="custom-li "
          onClick={[Function]}
        >
          Blog
        </li>
        <li
          className="custom-li selected"
          onClick={[Function]}
        >
          Company
        </li>
        <li
          className="custom-li "
          onClick={[Function]}
        >
          Careers
        </li>
        <li
          className="custom-li "
          onClick={[Function]}
        >
          Press
        </li>
      </ul>
    </nav>
  </header>
  <section
    className="hero"
    style={
      Object {
        "backgroundColor": "#f8f8f8",
        "padding": "40px",
      }
    }
  >
    <h1
      style={
        Object {
          "marginTop": "6px",
        }
      }
    >
      Expanding learning opportunities
    </h1>
  </section>
  <section
    className="origins"
  >
    <div
      className="origins-content"
    >
      <div
        className="origins-text"
      >
        <h2>
          Our Origins
        </h2>
        <p>
          Cursus was founded in 2020 by computer science instructor with a vision to provide anyone, anywhere with access to the worldâ€™s best education. Now many instructors put their courses online for anyone to take and taught more learners in a few months than they could over an entire lifetime in the classroom. Today, Cursus has expanded to reach more than 40 million people and 2,300 businesses around the world. On Cursus you can find online courses, instructors, and certificates from Cursus.
        </p>
      </div>
      <div
        className="origins-image"
      >
        <img
          alt="Company Origins"
          src="https://gambolthemes.net/html-items/cursus-new-demo/images/about/company.jpg"
        />
      </div>
    </div>
  </section>
  <section
    className="offices"
  >
    <h2>
      Our Offices
    </h2>
    <p>
      Cursus branches around the world
    </p>
    <div
      className="office-list"
    />
  </section>
</div>
`;